tap_github/client.py

```diff
@@ -211,17 +216,27 @@ def authed_get_all_pages(self, source, url, headers={}, stream="", should_skip_4
         """
         Fetch all pages of records and return them.
         """
+        url = self.prepare_url(url)
         while True:
             r = self.authed_get(source, url, headers, stream, should_skip_404)
             yield r
 
             # Fetch the next page if next found in the response.
-            if 'next' in r.links:
+            if 'next' in r.links and r.links['last'] != url:
```

Comment:
I'm not sure if this change is needed.

https://github.com/Automattic/nosara/blob/master/.teamcity/shared.py#L60

Why do you like `while True` so much? :trollface: 
Check the code above, isn't it easier to read?

Reply:
> Why do you like while True so much? :trollface:

It was already in forked code 😅 . I'll change.

> I'm not sure if this change is needed.

if you check this URL https://api.github.com/repositories/45717250/stargazers?page=1333
```
link: <https://api.github.com/repositories/45717250/stargazers?page=1332>; rel="prev", <https://api.github.com/repositories/45717250/stargazers?page=1334>; rel="next", <https://api.github.com/repositories/45717250/stargazers?page=1334>; rel="last", <https://api.github.com/repositories/45717250/stargazers?page=1>; rel="first"
```
The next page in the header is https://api.github.com/repositories/45717250/stargazers?page=1334 but if you try to open you will see this error:
```
{
  "message": "In order to keep the API fast for everyone, pagination is limited for this resource. Check the rel=last link relation in the Link response header to see how far back you can traverse.",
  "documentation_url": "https://docs.github.com/v3/#pagination"
}
```
So that's why I need to check if the URL is the last page to avoid this error.

Reply:
But I just saw a weird behavior, this error only happens if the last page is incomplete (if the result exceeds the 40k limit). Since I changed the results per page limit it's masking this error. Let me review it.

Reply:
> It was already in forked code

You have a club here! 😸